### ISSUE TITLE
Clarify the meaning of the number in the string type

### ISSUE THREAD

**Issue by lantw44** [ocpi/ocpi#418]
In the `CiString` type, we know it can only contain ASCII characters, so we can calculate the length by counting the number of bytes. However, in the `string` type, Unicode characters are allowed, but OCPI doesn't specify the way to calculate the string length. Does `string(255)` mean 255 bytes in UTF-8, 255 code points, or other things? Each programming language has its own definition of string length. I think OCPI should define the expected way to calculate string length, too.

**Comment by reinierl** [comment]
A string is a sequence of Unicode code points and therefore I would say `string(255)` means 255 Unicode code points. We can make that more explicit in future OCPI versions. Would that be a good solution for you too?

**Comment by lantw44** [comment]
Yes, counting the number of Unicode code points is what our implementation currently does.

**Comment by ahzf** [comment]
I would argue, that in more or less all cases in OCPI the limits on string lengths should be up to the implementations. We are no longer in the 1970ies and databases also optimize themselves these days. OCPI only limits its use cases e.g. too short fields for the calibration law in the past etc.pp
